WooCommerce/Classes/ViewRelated/Products/Edit Product/Cells/ImageAndTitleAndTextTableViewCell.swift

```diff
@@ -53,13 +53,24 @@ final class ImageAndTitleAndTextTableViewCell: UITableViewCell {
         let title: String?
     }
 
+    /// View model to replace TopLeftImageTableViewCell
+    struct TopLeftImageViewModel {
+        let icon: UIImage
+        let iconColor: UIColor?
+        let title: String
+        let isFootnoteStyle: Bool
```

Comment:
I thought to keep it simple for now and add more styles as we need them.

Another idea that just came to me is to define styles for this cell in a way similar to the standard UITableViewCell. We can have styles that are global for the cell vs styles that are attached to a particular view model.
For example, for this cell we can have the following styles: `default`, `switch`, `warning`, `footnote`, 'topLeftImage' (can do a better work on naming those :-) )

What do you think?

Reply:
sounds like a good idea! the current approach with different view models is a bit odd. if I understand correctly, the extra parameters in each view model could become associated values in each enum case like:

```swift
enum ImageAndTitleAndTextTableViewCell.Style {
  case default
  case switch(isSwitchOn: Bool, onSwitchChange: @escaping (_ isOn: Bool) -> Void)
  case warning
  case imageAndTitleOnly
}

extension ImageAndTitleAndTextTableViewCell {
  func updateUI(style: Style, title: String?, text: String?, textTintColor: UIColor? = nil, ...)
}
```

to replace these different view models: `ViewModel`, `SwitchableViewModel`, `WarningViewModel`, `TopLeftImageViewModel`

lemme know if this is different from what you thought!

Reply:
This is very similar to what I thought! I created an issue to keep track of it and so that it can be discussed there (https://github.com/woocommerce/woocommerce-ios/issues/3419)

Thank you!